### PR TITLE
new module: custom_gtffilter

### LINKED ISSUE #5963: new module: custom/gtffilter
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by nictru**
## PR checklist

The functionality has already been implemented in [rnaseq](https://github.com/nf-core/rnaseq/tree/master/modules/local/gtf_filter) and [scrnaseq](https://github.com/nf-core/scrnaseq/blob/master/modules/local/gtf_gene_filter.nf)

As I am planning to implement the same functionality in #circrna, I think it makes sense to extract this to a proper nf-core module

Closes #5963  

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [x] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [x] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [x] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Comment by nictru** [comment]
A matter for discussion would be the presence of `meta`: Both rnaseq and scrnaseq don't use it in their implementations, added it here for consistency

But in most use cases you will only have a single FASTA and a single GTF file

**Review by nvnieuwk — COMMENTED** [review]
Some small comments but LGTM overall

File: modules/nf-core/custom/gtffilter/main.nf [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    label 'process_single'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/python:3.9--1' :
+        'biocontainers/python:3.9--1' }"
+
+    input:
+    tuple val(meta), path(gtf)
+    path(fasta)
```
[the remark this answers is not in the document]
Comment (nictru) [reply modules/nf-core/custom/gtffilter/main.nf:12]: Ok yes, was not sure about metas here as I pointed out in [this comment](https://github.com/nf-core/modules/pull/5965#issuecomment-2222543078)

File: modules/nf-core/custom/gtffilter/templates/gtffilter.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+
+# Written by Olga Botvinnik with subsequent reworking by Jonathan Manning and Nico Trummer.
+# Released under the MIT license.
```
[the remark this answers is not in the document]
Comment (nictru) [reply modules/nf-core/custom/gtffilter/templates/gtffilter.py:4]: Is there an example for this in another module? I pretty much copied the license line from the RNA-Seq implementation, but I don't think this script is so sophisticated that it needs licensing

File: modules/nf-core/custom/gtffilter/main.nf (hunk 1 above)
[the remark this answers is not in the document]
Comment (nvnieuwk) [reply modules/nf-core/custom/gtffilter/main.nf]: Yeah it's just to stay consistent with all other modules :) This should simplify the implementation in the end

File: modules/nf-core/custom/gtffilter/templates/gtffilter.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nvnieuwk) [reply modules/nf-core/custom/gtffilter/templates/gtffilter.py:4]: The `gprofiler2/gost` module does that. There are indeed a lot of modules that do it like you did, but I think in the case of licenses that it's better to be complete about it

**Review by nvnieuwk — APPROVED** [review]
LGTM!

File: modules/nf-core/custom/gtffilter/templates/gtffilter.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nictru) [reply modules/nf-core/custom/gtffilter/templates/gtffilter.py]: Added it!

**Pull request merged**